contracts/voting-verifier/src/client.rs

```diff
@@ -0,0 +1,212 @@
+        self.client
+            .execute(&ExecuteMsg::VerifyMessages { messages })
+    }
+
+    pub fn vote(&self, poll_id: PollId, votes: Vec<Vote>) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::Vote { poll_id, votes })
+    }
+
+    pub fn end_poll(&self, poll_id: PollId) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::EndPoll { poll_id })
+    }
+
+    pub fn verify_worker_set(
+        &self,
+        message_id: nonempty::String,
+        new_operators: Operators,
+    ) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::VerifyWorkerSet {
+            message_id,
+            new_operators,
+        })
+    }
+
+    pub fn update_voting_threshold(&self, new_voting_threshold: MajorityThreshold) -> WasmMsg {
+        self.client.execute(&ExecuteMsg::UpdateVotingThreshold {
+            new_voting_threshold,
+        })
+    }
+
+    pub fn poll(&self, poll_id: PollId) -> Result<Poll> {
+        self.client
+            .query(&QueryMsg::GetPoll { poll_id })
+            .change_context_lazy(|| Error::QueryVotingVerifier(self.client.address.clone()))
+    }
+
+    pub fn messages_status(
+        &self,
+        messages: Vec<Message>,
+    ) -> Result<Vec<(CrossChainId, VerificationStatus)>> {
+        match messages.as_slice() {
+            [] => Ok(vec![]),
+            _ => self
+                .client
+                .query(&QueryMsg::GetMessagesStatus { messages })
+                .change_context_lazy(|| Error::QueryVotingVerifier(self.client.address.clone())),
+        }
+    }
+
+    pub fn worker_set_status(&self, new_operators: Operators) -> Result<VerificationStatus> {
+        self.client
+            .query(&QueryMsg::GetWorkerSetStatus { new_operators })
+            .change_context_lazy(|| Error::QueryVotingVerifier(self.client.address.clone()))
+    }
+
+    pub fn current_threshold(&self) -> Result<MajorityThreshold> {
+        self.client
+            .query(&QueryMsg::GetCurrentThreshold)
+            .change_context_lazy(|| Error::QueryVotingVerifier(self.client.address.clone()))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use axelar_wasm_std::{operators::Operators, Threshold, VerificationStatus};
+    use connection_router_api::{Message, CHAIN_NAME_DELIMITER};
+    use cosmwasm_std::{
+        from_binary,
+        testing::{mock_dependencies, mock_env, mock_info, MockQuerier},
+        Addr, DepsMut, QuerierWrapper, Uint256, Uint64, WasmQuery,
+    };
+
+    use crate::{
+        contract::{instantiate, query},
+        msg::{InstantiateMsg, QueryMsg},
+        Client,
+    };
+
+    #[test]
+    fn query_messages_status() {
+        let (querier, _, addr) = setup();
+        let client: Client = client::Client::new(QuerierWrapper::new(&querier), addr).into();
+
+        let msg_1 = Message {
+            cc_id: format!("eth{}0x1234", CHAIN_NAME_DELIMITER)
+                .as_str()
+                .parse()
+                .unwrap(),
```

Comment:
Can we use the `HexTxHashAndEventIndex` struct to construct a properly formed message id? Hard coded strings like this are such a pain if they ever need to be changed. And it's likely this will need to be changed, because the tx hash is not the right length, and there's no event index.